src/EventSubscriber/XLocationIdResponseSubscriber.php

```diff
@@ -0,0 +1,68 @@
+<?php
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\PlatformHttpCacheBundle\EventSubscriber;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\FilterResponseEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+/**
+ * Rewrites the X-Location-Id HTTP header.
+ *
+ * This is a BC layer for custom controllers (including REST server) still
+ * using X-Location-Id header which is now deprecated. For
+ * full value of tagging, see docs/using_tags.md for how to take advantage of the
+ * system.
+ */
+class XLocationIdResponseSubscriber implements EventSubscriberInterface
+{
+    const LOCATION_ID_HEADER = 'X-Location-Id';
+
+    /**
+     * @var string
+     */
+    private $tagHeader;
+
+    public function __construct($tagHeader)
+    {
+        $this->tagHeader = $tagHeader;
+    }
+
+    public static function getSubscribedEvents()
+    {
+        return [KernelEvents::RESPONSE => ['rewriteCacheHeader', -5]];
+    }
+
+    public function rewriteCacheHeader(FilterResponseEvent $event)
+    {
+        $response = $event->getResponse();
+        if (!$response->headers->has(static::LOCATION_ID_HEADER)) {
+            return;
+        }
+
+        @trigger_error(
+            'X-Location-Id is no longer preferred way to tag content responses, see ezplatform-http-cache/docs/using_tags.md',
+            E_USER_DEPRECATED
+        );
+
+        // Map the tag, even if not officially supported, handle comma separated values as was possible with Varnish
+        $tags = array_map(
+            function ($id) {
+                return 'location-' . trim($id);
+            },
+            explode(',', $response->headers->get(static::LOCATION_ID_HEADER))
+        );
+
+        if ($response->headers->has($this->tagHeader)) {
+            $tags = array_merge($response->headers->get($this->tagHeader, null, false) , $tags);
+        }
+
+        // @todo we need to use abstract tag writer to also be able to support Fastly
+        // FOS has some stuff around this in 2.x but not in a good way in 1.x
+        $response->headers->set($this->tagHeader, array_unique($tags));
```

Comment:
I don't like this. Now we have logic for setting xkey both here and in https://github.com/ezsystems/ezplatform-http-cache/blob/a4ed9b6b8d9a9dfc4a0f92e8a1e1de56a877d9c3/src/ResponseConfigurator/ConfigurableResponseCacheConfigurator.php#L63

This should be done in one place.
I fancy to replace the whole ConfigurableResponseCacheConfigurator with FOS\HttpCacheBundle\Handler\TagHandler, but that is not quite straight forward.
So instead I am thinking just adding a TagHandler and make sure ConfigurableResponseCacheConfigurator  and XLocationIdResponseSubscriber  uses that one for setting the headers. What do you say?

Reply:
Okay, Adding our own TagHandler based on FOS\HttpCacheBundle\Handler was not a good idea for several reasons.
One was that we are not able to run it's constructor without also adding a CacheInvalidator ( kinda replaces our PurgeClient) and that it's tag handling doesn't really suit our needs in regards to either xkey nor fastly.

Reply:
I'll then create our abstraction for this

Reply:
thread marked as obsolete :)